src/main/java/com/sngular/kloadgen/sampler/schemaregistry/impl/ApicurioSchemaRegistry.java

```diff
@@ -0,0 +1,174 @@
+          SchemaParser parserJson = new JsonSchemaParser();
+          schema.setSchema(parserJson.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
+        } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(searchedArtifactType)) {
+          SchemaParser parserProtobuf = new ProtobufSchemaParser();
+          schema.setSchema(parserProtobuf.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
+        } else {
+          throw new KLoadGenException(String.format("Schema type not supported %s", searchedArtifactType));
+        }
+        schema.setType(searchedArtifactType);
+        return schema;
+      }
+    } catch (IOException e) {
+      throw new KLoadGenException(e);
+    }
+  }
+
+  public Object getSchemaBySubjectAndId(String subjectName, Object metadata) {
+    Object schema = null;
+    final ArtifactMetaData schemaMetadata = (ArtifactMetaData) metadata;
+
+    try {
+      InputStream inputStream = this.schemaRegistryClient.getContentByGlobalId(schemaMetadata.getGlobalId());
+      String result = IOUtils.toString(inputStream, String.valueOf(StandardCharsets.UTF_8));
+
+      String searchedArtifactType = schemaMetadata.getType().toString();
+      if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(searchedArtifactType)) {
+        SchemaParser parserAvro = new AvroSchemaParser(null);
+        schema = parserAvro.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>());
+      } else if (SchemaTypeEnum.JSON.name().equalsIgnoreCase(searchedArtifactType)) {
+        SchemaParser parserJson = new JsonSchemaParser();
+        schema = parserJson.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>());
+      } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(searchedArtifactType)) {
+        SchemaParser parserProtobuf = new ProtobufSchemaParser();
+        schema = parserProtobuf.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>());
+      } else {
+        throw new KLoadGenException(String.format("Schema type not supported %s", searchedArtifactType));
+      }
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+    return schema;
+  }
+
+  private SearchedArtifact getLastestSearchedArtifact(final String subjectName) {
+    boolean found = false;
+    SearchedArtifact best = null;
+    final Comparator<SearchedArtifact> comparator = Comparator.comparing(SearchedArtifact::getModifiedOn);
+    for (SearchedArtifact artifact : this.schemaRegistryClient.searchArtifacts(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_DEFAULT_GROUP, null, null, null, null, null, null, null,
+                                                                               null,
+                                                                               null, null).getArtifacts()) {
+      if (artifact.getName().equals(subjectName) && artifact.getState().equals(ArtifactState.ENABLED)) {
```

Comment:
A few things:
To compare strings, it would be better `ArtifactState.ENABLED.equals(artifact.getState())`
And due to there is no `else` in both of them, we can join with `and`
```
if (artifact.getName().equals(subjectName) && artifact.getState().equals(ArtifactState.ENABLED) && (!found || comparator.compare(artifact, best) > 0))
```